### PR TITLE
Added test scenarios for Hugepage and Hugepage pinning

### PR DESCRIPTION
**Pull request by balamuruhans**
Added test scenarios for Hugepage and Hugepage pinning
    
    This patch enables hugepage, hugepage pinning for covering following scenarios
    
    1. migration with only HP
    2. migration with numa node and HP
    3. migration with numa node and HP pin
    4. migration with numa pinning and HP
    5. migration with numa pinning and HP pin

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 1]
```diff
@@ -169,6 +171,103 @@ def create_numa(vcpu, max_mem, max_mem_unit):
...
+        :param hp_unit: unit of HP size
+        :param hp_node: number of numa nodes to be HP pinned
+        :param pin: flag to pin HP with guest numa or not
+        :param node_list: Numa node list
+        :param host_hp_size: size of the HP to pin with guest numa
+        :param numa_pin: flag to numa pin
+        :return: list of page tag dictionary for HP pin
+        """
+        dest_machine = params.get("migrate_dest_host")
+        command = "ssh root@%s 'cat /proc/meminfo | grep HugePages_Free' | awk '{print $2}'" % dest_machine
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: this assumes ssh connection without password, it'll fail if without setting, and for remote session, you could use remote module remote.wait_for_login()

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 2]
```diff
@@ -303,6 +404,30 @@ def create_numa(vcpu, max_mem, max_mem_unit):
             raise error.TestNAError("NUMA memory tuning in preferred mode only"
                                     " supports single node")
 
+    # To check if Hugepage supported and configure
+    if enable_HP or enable_HP_pin:
+        try:
+            # For guest
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: This is not for check guest support, it's for check host and set related part on host.
So you could put this "for guest" and "for host" part together, and only use test_setup module for host setting.

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py (hunk 1 above)
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: @waynesun09 yeah, I used remote module remote.wait_for_login() for all remote session commands.

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py (hunk 2 above)
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: @waynesun09, changed as per your comments

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 3]
```diff
@@ -169,6 +171,106 @@ def create_numa(vcpu, max_mem, max_mem_unit):
...
+        if (cmd_output[0] == 0):
+            dest_HP_free = cmd_output[1].strip('HugePages_Free:').strip()
+        else:
+            raise error.TestNAError("HP not supported/configured")
+        hp_list = []
+
+        # setting hugepages in destination machine here as remote ssh
+        # configuration is done
+        hugepage_assign(
+            str(no_of_HPs), target_ip=dest_machine, user=server_user, password=server_pwd)
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: better switch line and keep line length under 80.
also at following codes.

Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: @waynesun09 sure, I will change accordingly

**Comment by balamuruhans** [comment]
@waynesun09 I have changed the code to keep line length under 80 and also fixed some issues observed while testing

**Comment by balamuruhans** [comment]
@waynesun09, Please help me in merging this PR.

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 4]
```diff
@@ -169,6 +171,113 @@ def create_numa(vcpu, max_mem, max_mem_unit):
...
+        """
+        command = ""
+        if node == '':
+            if target_ip == '':
+                utils_memory.set_num_huge_pages(int(hp_num))
+            else:
+                command = "echo %s > /proc/sys/vm/nr_hugepages" % (hp_num)
+        else:
+            command = "echo %s > /sys/devices/system/node/node%s/hugepages/\
+hugepages-%skB/nr_hugepages" % (hp_num, str(node), hp_size)
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: It's better split it here:

command = "echo %s > /sys/devices/system/node/node" % hp_num
command += "%s/hugepages/hugepages-%skB/nr_hugepages" % (str(node), hp_size)

Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: @waynesun09 sure

**Comment by balamuruhans** [comment]
@waynesun09 done

**Comment by waynesun09** [comment]
@balamuruhans Thanks, merge now.

**Comment by balamuruhans** [comment]
@waynesun09 Thank you

**Pull request merged**